pkg/redirectpolicy/manager.go

```diff
@@ -449,10 +449,12 @@ func (rpm *Manager) notifyPolicyBackendDelete(config *LRPConfig, frontendMapping
 			log.WithError(err).Errorf("Local redirect service for policy (%v)"+
 				" with frontend (%v) not deleted", config.id, frontendMapping.feAddr)
 		}
-		if restored := rpm.svcCache.EnsureService(*config.serviceID, lock.NewStoppableWaitGroup()); restored {
-			log.WithFields(logrus.Fields{
-				logfields.K8sSvcID: *config.serviceID,
-			}).Info("Restored service")
+		if config.lrpType == lrpConfigTypeSvc {
```

Comment:
This doesn't actually solve the nil deref. We could go for:

```suggestion
		if config.lrpType == lrpConfigTypeSvc && config.serviceID != nil {
```

or:

```suggestion
		if config.lrpType == lrpConfigTypeSvc {
			if config.serviceID == nil {
				// log an error
			}
```

Reply:
The field will not be `nil` for a service matcher LRP by contract - https://github.com/cilium/cilium/blob/master/pkg/k8s/apis/cilium.io/v2/clrp_types.go#L91

Reply:
Sure thing, but do we prefer a panic+crash in case it's nil due to someone introducing a bug?

Reply:
It won't be `nil`, check my previous comment. Regardless, I don't want to pepper the entire code with such redundant checks. I would rather have such code paths validated by CI/unit tests (it's in my TODO list). :)